cms/djangoapps/contentstore/rest_api/v2/views/tests/test_downstreams.py

```diff
@@ -820,139 +1056,91 @@ def test_200_all_component_downstreams_for_a_course(self):
-        Does the happy path work?
-        """
-        self.client.login(username="superuser", password="password")
-        response = self.call_api(str(self.another_course.id))
-        assert response.status_code == 200
-        data = response.json()
-        expected = [{
-            'upstream_context_title': 'Test Library 1',
-            'upstream_context_key': self.library_id,
-            'ready_to_sync_count': 0,
-            'total_count': 3,
-            'last_published_at': self.now.strftime('%Y-%m-%dT%H:%M:%S.%fZ'),
-        }]
-        self.assertListEqual(data, expected)
-        response = self.call_api(str(self.course.id))
-        assert response.status_code == 200
-        data = response.json()
-        expected = [{
-            'upstream_context_title': 'Test Library 1',
-            'upstream_context_key': self.library_id,
-            'ready_to_sync_count': 2,
-            'total_count': 5,
-            'last_published_at': self.now.strftime('%Y-%m-%dT%H:%M:%S.%fZ'),
-        }]
-        self.assertListEqual(data, expected)
+        # Publish Section and component/unit that has the same section as top-level parent
+        self._update_container(self.top_level_section_id, display_name="Section 3")
+        self._publish_container(self.top_level_section_id)
+        self._set_library_block_olx(self.video_lib_id_2, "<video><b>Hello world!</b></video>")
+        self._publish_library_block(self.video_lib_id_2)
+        self._update_container(self.top_level_subsection_id, display_name="Subsection 3")
+        self._publish_container(self.top_level_subsection_id)
 
-
-class GetContainerUpstreamViewTest(
-    _BaseDownstreamViewTestMixin,
-    SharedModuleStoreTestCase,
-):
-    """
-    Test that `GET /api/v2/contentstore/downstream-containers?...` returns list of links based on the provided filter.
-    """
-    def call_api(
-        self,
-        course_id: str | None = None,
-        ready_to_sync: bool | None = None,
-        upstream_container_key: str | None = None,
-    ):
-        data = {}
-        if course_id is not None:
-            data["course_id"] = str(course_id)
-        if ready_to_sync is not None:
-            data["ready_to_sync"] = str(ready_to_sync)
-        if upstream_container_key is not None:
-            data["upstream_container_key"] = str(upstream_container_key)
-        return self.client.get("/api/contentstore/v2/downstream-containers/", data=data)
-
-    def test_200_all_container_downstreams_for_a_course(self):
-        """
-        Returns all container links for given course
-        """
-        self.client.login(username="superuser", password="password")
-        response = self.call_api(course_id=self.course.id)
+        response = self.call_api(
+            ready_to_sync=True,
+            item_type="all",
+            use_top_level_parents=True,
+        )
         assert response.status_code == 200
         data = response.json()
+        self.assertEqual(data["count"], 3)
```

Comment:
After making the above suggested change, this increases to 4, which seems correct as we are also making a change to top level subsection in line 1096 no?